homeassistant/components/soma/__init__.py

```diff
@@ -99,3 +105,38 @@ def device_info(self) -> DeviceInfo:
             manufacturer="Wazombi Labs",
             name=self.name,
         )
+
+    def set_position(self, position: int):
+        """Set the current device position."""
+        self.current_position = position
+        self.schedule_update_ha_state()
+
+    async def async_get_state_from_api(self) -> dict:
+        """Get the latest data from the API."""
+        responseFromApi = {}
+        try:
+            response = await self.hass.async_add_executor_job(
+                self.api.get_shade_state, self.device["mac"]
+            )
+
+            if not self.api_is_available:
+                self.api_is_available = True
+                _LOGGER.info("Connection to SOMA Connect succeeded")
+
+            if not is_api_response_success(response):
+                if self.is_available:
+                    self.is_available = False
+                    _LOGGER.warning(
+                        f'Device is unreachable ({self.name}). Error while fetching the state: {response["msg"]}'
+                    )
+            else:
+                if not self.is_available:
+                    self.is_available = True
+                    _LOGGER.info(f"Device {self.name} is now reachable")
+                responseFromApi = response
```

Comment:
Please move code out of the `try` block that isn't expected to throw `RequestException` (probably into a `try/except/else`)

Reply:
Good catch!
Fixed